Sources/Interaction/Widgets/ImageCroppingRegionsRepresentation/index.js

```diff
@@ -0,0 +1,390 @@
+        centerVerts.set([slicePos, yPUpper, zPUpper], 6);
+        centerVerts.set([slicePos, yPLower, zPUpper], 9);
+        break;
+      case Orientation.XZ:
+        verts.set([xBMin, slicePos, zBMin], 0);
+        verts.set([xPLower, slicePos, zBMin], 3);
+        verts.set([xPLower, slicePos + 0.01, zPLower], 6);
+        verts.set([xBMin, slicePos, zPLower], 9);
+        verts.set([xPUpper, slicePos + 0.01, zBMin], 12);
+        verts.set([xBMax, slicePos, zBMin], 15);
+        verts.set([xBMax, slicePos + 0.01, zPLower], 18);
+        verts.set([xPUpper, slicePos, zPLower], 21);
+        verts.set([xPUpper, slicePos + 0.01, zPUpper], 24);
+        verts.set([xBMax, slicePos, zPUpper], 27);
+        verts.set([xBMax, slicePos, zBMax], 30);
+        verts.set([xPUpper, slicePos, zBMax], 33);
+        verts.set([xBMin, slicePos, zPUpper], 36);
+        verts.set([xPLower, slicePos, zPUpper], 39);
+        verts.set([xPLower, slicePos + 0.01, zBMax], 42);
+        verts.set([xBMin, slicePos, zBMax], 45);
+
+        centerVerts.set([xPLower, slicePos, zPLower], 0);
+        centerVerts.set([xPUpper, slicePos, zPLower], 3);
+        centerVerts.set([xPUpper, slicePos, zPUpper], 6);
+        centerVerts.set([xPLower, slicePos, zPUpper], 9);
+        break;
+      case Orientation.XY:
+        verts.set([xBMin, yBMin, slicePos], 0);
+        verts.set([xPLower, yBMin, slicePos], 3);
+        verts.set([xPLower, yPLower, slicePos + 0.01], 6);
+        verts.set([xBMin, yPLower, slicePos], 9);
+        verts.set([xPUpper, yBMin, slicePos + 0.01], 12);
+        verts.set([xBMax, yBMin, slicePos], 15);
+        verts.set([xBMax, yPLower, slicePos + 0.01], 18);
+        verts.set([xPUpper, yPLower, slicePos], 21);
+        verts.set([xPUpper, yPUpper, slicePos + 0.01], 24);
+        verts.set([xBMax, yPUpper, slicePos], 27);
+        verts.set([xBMax, yBMax, slicePos], 30);
+        verts.set([xPUpper, yBMax, slicePos], 33);
+        verts.set([xBMin, yPUpper, slicePos + 0.01], 36);
+        verts.set([xPLower, yPUpper, slicePos], 39);
+        verts.set([xPLower, yBMax, slicePos + 0.01], 42);
+        verts.set([xBMin, yBMax, slicePos], 45);
+
+        centerVerts.set([xPLower, yPLower, slicePos], 0);
+        centerVerts.set([xPUpper, yPLower, slicePos], 3);
+        centerVerts.set([xPUpper, yPUpper, slicePos], 6);
+        centerVerts.set([xPLower, yPUpper, slicePos], 9);
+        break;
+      default:
+      // noop
+    }
+
+    // This requires updating whenever geometry updates
+    updateCenterOpacity();
+
+    model.regionPolyData.modified();
+    model.centerRegionPolyData.modified();
+
+    publicAPI.invokePlanesPositionChanged();
+    publicAPI.modified();
+  };
+
+  publicAPI.getBounds = () => model.initialBounds;
+
+  publicAPI.buildRepresentation = () => {
+    if (model.renderer) {
+      if (!model.placed) {
+        model.validPick = 1;
+        model.placed = 1;
+      }
+      publicAPI.modified();
```

Comment:
always?

Reply:
I'm not too sure with buildRepresentation, since that's called by AbstractWidget. Other buildRepresentations call modified(), so I went along with it, even though I build the representation in the constructor rather than here. Maybe I can move that code to buildRepresentation now that I think about it...

Reply:
You know the code better than I do. I let you do another path to clean it up. I was just mentioning thing that might be important such the modified only when truly required.

Reply:
I'll do another pass. I saw LineRepresentation constructed the model inside extend(), which I found odd so I opted to put the construction elsewhere.

Reply:
Yes model should not be filled inside extend() unless it is 3 lines...